main/src/main/scala/sbt/internal/server/BuildServerReporter.scala

```diff
@@ -188,6 +204,10 @@ final class BuildServerReporterImpl(
   }
 }
 
+object BuildServerReporterImpl {
+  private val hasAnyProblems = mutable.Set[Path]()
```

Comment:
For 1million of paths `hasAnyProblems` will have ~90MB. Checked using https://gist.github.com/kpodsiad/2d67e15d7c14d78b1711bc1e6cb36dcc